src/blur.js

```diff
@@ -0,0 +1,64 @@
+function blurTransfer(V, r, n, vertical) {
+  const [source, target] = V,
+    m = Math.floor(source.length / n),
+    w = (r << 1) + 1,
+    w1 = 1 / w,
+    ki = vertical ? m : 1,
+    kj = vertical ? 1 : n,
+    W = w * ki,
+    R = r * ki;
+  for (let j = 0; j < m; ++j) {
+    const k0 = kj * j,
+      kn = k0 + ki * (n - 1);
+    for (let i = 0, sr = w * source[k0]; i < n + r; ++i) {
+      const k = ki * i + kj * j;
+      sr += source[Math.min(k, kn)] - source[Math.max(k - W, k0)];
+      target[Math.max(k - R, k0)] = sr * w1;
+    }
+  }
+  V.reverse();
+}
+
+export default function blur() {
+  var radius = 5,
```

Comment:
Use of `var` vs. `let`/`const` is inconsistent within this file.

Reply:
Yeah it's going to be inconsistent all over the d3 code base for a while. But for "new" code I suppose we'll want ES6?